browser-api/src/main/java/com/duckduckgo/app/notification/model/SchedulableNotificationPlugin.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.notification.model
+
+interface SchedulableNotificationPlugin {
```

Comment:
This was created to be able to have notifications in different modules. The idea is to create one plugin per notification and then use them to do different things defined in the module itself (i.e. specific action when a notification is shown, etc)

I created https://app.asana.com/0/414730916066338/1201842645469204 to write about this.